### ISSUE TITLE
`UndefinedMagicMethod` for all @method declarations on `Illuminate\Support\Facades\Request`

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#886]
**Describe the bug**

Calls to `Illuminate\Support\Facades\Request::*` are reported as `UndefinedMagicMethod` even when the method is declared via `@method` in the facade docblock. This affects every `@method` on this facade (`input`, `method`, `route`, etc.), not a single missing tag.

`Illuminate\Support\Facades\Cache::get(...)` on the same run does **not** error, so the regression is specific to the `Request` facade (or its resolution path through the plugin), not to `@method` parsing in general.

`AppFacadeRegistrationHandler` explicitly skips `Illuminate\` facades (line 31-33 of its class docblock), and there are no relevant project stubs, so the error originates either from Psalm core failing to consult the facade's `@method` catalogue, or from the plugin's `FacadeMapProvider`/handler path interfering despite the skip.

**Code to reproduce**

```php
<?php
use Illuminate\Support\Facades\Request;
use Illuminate\Support\Facades\Cache;

function a(): mixed { return Request::input('k'); } // ERROR: UndefinedMagicMethod
function b(): string { return Request::method(); }  // ERROR: UndefinedMagicMethod
function c(): mixed { return Request::route('r'); } // ERROR: UndefinedMagicMethod
function d(): mixed { return Cache::get('k'); }     // OK
```

Run with cache cleared (`rm -rf .cache/psalm && vendor/bin/psalm <file>`) to rule out stale storage.

`Illuminate\Support\Facades\Request` declares all three in the framework facade docblock, e.g.:

```
@method static mixed input(string|null $key = null, mixed $default = null)
```

**Impacted Versions**

```text
laravel/framework        v12.58.0
psalm/plugin-laravel     dev-master (22af382, 2026-05-01)
vimeo/psalm              7.0.0-beta19
PHP                      8.4.20
```

**Psalm config**

Minimal `psalm.xml` reproducing this only needs the Laravel plugin enabled:

```xml
<psalm errorLevel="1" resolveFromConfigFile="true" cacheDirectory="./.cache/psalm">
    <plugins>
        <pluginClass class="Psalm\LaravelPlugin\Plugin"/>
    </plugins>
    <projectFiles>
        <directory name="app"/>
        <ignoreFiles><directory name="vendor"/></ignoreFiles>
    </projectFiles>
</psalm>
```

**Additional context**

- `AppFacadeRegistrationHandler` skips `Illuminate\*` facades intentionally; `FacadeMapProvider::init()` builds the alias map by calling `$facadeClass::getFacadeRoot()`. For the `Request` facade the accessor is `'request'`, which under Testbench may throw a `BindingResolutionException` (no current request bound). It is silently caught at line 68 and the facade is skipped from the map. If `Cache` works because its container binding resolves cleanly, the divergence may live there.
- The same calls are present in real codebases and are currently being papered over via `psalm-baseline.xml` rather than fixed.
- Happy to open a PR once we agree on where the fix should live (plugin handler bridging `Request` to `Illuminate\Http\Request`, or a Psalm core question about pseudo_static_methods on this facade).

**Comment by alies-dev** [comment]
**Root cause found — this is not a plugin bug.**

The `Illuminate\Support\Facades\Request` docblock at line 178 (Laravel 12.58.0) contains a malformed `@method` tag with an unbalanced parenthesis:

```
* @method static \BackedEnum|(\BackedEnum|null enum(string $key, string $enumClass, \BackedEnum|null $default = null)
```

The `(\BackedEnum|null` opens a grouping paren that is never closed. Psalm's docblock parser aborts on this line, which drops every `@method` tag on the facade — that's why `input`, `method`, `route` etc. all uniformly disappear, while `Cache::*` (different facade, intact docblock) is unaffected.

**Origin:** Laravel commit `558dd712e` ("Update facade docblocks", Taylor Otwell, 2026-01-28). First tagged in `v12.50.0`. Present on both `12.x` and `13.x` branches today. The auto-generator for facade docblocks took the `enum()` source signature (templated `TEnum|TDefault` where `TDefault of TEnum|null`) and emitted the bounded `TDefault` as `(\BackedEnum|null)` for grouping, but dropped the closing paren.

**Reproducible without the plugin loaded** — purely a Psalm-core parsing failure on broken upstream input.

**Suggested next steps:**
1. Fix at the source: PR against `laravel/framework` to balance the paren (or remove the redundant grouping entirely, since `\BackedEnum|(\BackedEnum|null)` reduces to `\BackedEnum|null`).
2. Defensive: open a Psalm-core issue asking the docblock parser to recover from a single malformed `@method` rather than dropping the whole catalogue.

Closing as not-a-plugin-bug seems appropriate — happy to do that once acknowledged, or leave it open as a tracking link to the upstream fix.

**Comment by alies-dev** [comment]
commit with a bug: https://github.com/laravel/framework/commit/558dd712e

**Comment by alies-dev** [comment]
affected Laravel versions:
[v13.8.0](https://github.com/laravel/framework/releases/tag/v13.8.0)
[v13.7.0](https://github.com/laravel/framework/releases/tag/v13.7.0)
[v13.6.0](https://github.com/laravel/framework/releases/tag/v13.6.0)
[v13.5.0](https://github.com/laravel/framework/releases/tag/v13.5.0)
[v13.4.0](https://github.com/laravel/framework/releases/tag/v13.4.0)
[v13.3.0](https://github.com/laravel/framework/releases/tag/v13.3.0)
[v13.2.0](https://github.com/laravel/framework/releases/tag/v13.2.0)
[v13.1.1](https://github.com/laravel/framework/releases/tag/v13.1.1)
[v13.1.0](https://github.com/laravel/framework/releases/tag/v13.1.0)
[v13.0.0](https://github.com/laravel/framework/releases/tag/v13.0.0)
[v12.58.0](https://github.com/laravel/framework/releases/tag/v12.58.0)
[v12.57.0](https://github.com/laravel/framework/releases/tag/v12.57.0)
[v12.56.0](https://github.com/laravel/framework/releases/tag/v12.56.0)
[v12.55.1](https://github.com/laravel/framework/releases/tag/v12.55.1)
[v12.55.0](https://github.com/laravel/framework/releases/tag/v12.55.0)
[v12.54.1](https://github.com/laravel/framework/releases/tag/v12.54.1)
[v12.54.0](https://github.com/laravel/framework/releases/tag/v12.54.0)
[v12.53.0](https://github.com/laravel/framework/releases/tag/v12.53.0)
[v12.52.0](https://github.com/laravel/framework/releases/tag/v12.52.0)
[v12.51.0](https://github.com/laravel/framework/releases/tag/v12.51.0)
[v12.50.0](https://github.com/laravel/framework/releases/tag/v12.50.0)
